libbeat/esleg/eslegclient/connection.go

```diff
@@ -436,7 +437,8 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 	}
 
 	if conn.APIKey != "" {
-		req.Header.Add("Authorization", "ApiKey "+conn.APIKey)
+		encoded := base64.StdEncoding.EncodeToString([]byte(conn.APIKey))
+		req.Header.Add("Authorization", "ApiKey "+encoded)
```

Comment:
would it make sense to cache the result in a private variable like `conn.encodedAPIKey` ?

Reply:
Yes, it would. No need to encode it each time we make a request to ES. Thanks for the suggestion, will implement!

Reply:
Done in c3e5b43.